### PR TITLE
EHN: adding erf in paddle.tensor.tensor.Tensor

### PR DESCRIPTION
**Pull request by MuhammadNizamani**
adding function erf in paddle.tensor.tensor.Tensor from issue #16552

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 1]
```diff
@@ -727,3 +727,36 @@ def test_paddle_tanh(
...
     )
+
+
+# erf
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="erf",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:738]:
```suggestion
        available_dtypes=helpers.get_dtypes("valid"),
```
Let's test with valid dtypes

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 2]
```diff
@@ -138,6 +138,10 @@ def cos(self, name=None):
     def exp(self, name=None):
         return ivy.exp(self._ivy_array)
 
+    @with_supported_dtypes({"2.4.2 and below": ("float32", "float64")}, "paddle")
```
Comment (ZiadAmerr) [inline ivy/functional/frontends/paddle/tensor/tensor.py:141]: Is there any reference to these dtypes being unsupported?

**Review by ZiadAmerr — COMMENTED** [review]
Just this final comment and we're ready to merge :)

**Comment by ZiadAmerr** [comment]
Please request my review when your PR is ready for merging so that I can get a notification and merge it more quickly, thanks a lot @MuhammadNizamani :)

File: ivy/functional/frontends/paddle/tensor/tensor.py (hunk 2 above)
Comment (MuhammadNizamani) [reply ivy/functional/frontends/paddle/tensor/tensor.py:141]: these are supported here you can see in the paddle docs
paramet

**Comment by MuhammadNizamani** [comment]
@ZiadAmerr  is there any thing that I should change

**Review by ZiadAmerr — APPROVED** [review]
Amazing work @MuhammadNizamani! We will just wait for the CI tests to pass and we're good to go :)

**Comment by MuhammadNizamani** [comment]
@ZiadAmerr  why these check are getting failed ? if this i should convert this `available_dtypes=helpers.get_dtypes("valid")`,  into `available_dtypes=helpers.get_dtypes("float"),`

**Comment by ZiadAmerr** [comment]
We should always test with all the valid dtypes for that particular function. :)

**Comment by MuhammadNizamani** [comment]
@zaeemansari70  so why these checks are being failed?

**Comment by MuhammadNizamani** [comment]
@ZiadAmerr  it is ready to merge?

**Comment by ZiadAmerr** [comment]
Are the tests passing locally?

**Comment by MuhammadNizamani** [comment]
Should I run pytest or something else?

**Comment by ZiadAmerr** [comment]
Yes pytest on your function only and with detailed output flag

**Comment by MuhammadNizamani** [comment]
Ok I will run pytest on it .

**Comment by MuhammadNizamani** [comment]
@ZiadAmerr i check out this pic 
erfpyest

**Comment by ZiadAmerr** [comment]
This is a very common problem facing the testing where `framework` is `None`. You can check out the ivy-tests channel on discord for a solution to that. :)

**Comment by MuhammadNizamani** [comment]
@ZiadAmerr all 3 test pass for erf check the following image 
erfpasstest

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Review by ZiadAmerr — APPROVED** [review]
Amazing work!

**Comment by ZiadAmerr** [comment]
LGTM, thanks @MuhammadNizamani! :)

**Pull request merged**